kit/inputs/src/UIMaterialTextView/UIMaterialTextViewFloating.tsx

```diff
@@ -0,0 +1,179 @@
+import * as React from 'react';
+import { TextInput, View } from 'react-native';
+
+import { useHover } from '@tonlabs/uikit.controls';
+import { UILayoutConstant } from '@tonlabs/uikit.layout';
+import { makeStyles, useTheme, Theme, ColorVariants } from '@tonlabs/uikit.themes';
+import Animated, { interpolate, useAnimatedStyle, Layout } from 'react-native-reanimated';
+import { UITextView, useFocused, useUITextViewValue } from '../UITextView';
+
+import { useMaterialTextViewChildren } from './useMaterialTextViewChildren';
+
+import { FloatingLabel } from './FloatingLabel';
+import type { UIMaterialTextViewProps, UIMaterialTextViewRef } from './types';
+import { useExtendedRef } from './useExtendedRef';
+import { useAutogrow } from './useAutogrow';
+import { UIMaterialTextViewComment } from './UIMaterialTextViewComment';
+import { useExpandingValue } from './useExpandingValue';
+
+// @inline
+const POSITION_FOLDED: number = 0;
+// @inline
+const POSITION_EXPANDED: number = 1;
+
+// EXPANDED_INPUT_OFFSET = UILayoutConstant.contentOffset - UILayoutConstant.contentInsetVerticalX2
```

Comment:
What about this comment? Maybe kill it if it's not needed?

Reply:
It says here how this constant is calculated